bundles/org.eclipse.swt/Eclipse SWT Custom Widgets/common/org/eclipse/swt/custom/CTabFolderRenderer.java

```diff
@@ -318,9 +318,14 @@ protected Point computeSize (int part, int state, GC gc, int wHint, int hHint) {
 					Image image = item.getImage();
 					if (image != null && !image.isDisposed()) {
 						Rectangle bounds = image.getBounds();
-						if ((state & SWT.SELECTED) != 0 || parent.showUnselectedImage) {
-							width += bounds.width;
+						//If image is not visible, tab width should not include image width
+						//Additionally, tab width is increased by standard padding (to add padding to the text) 
+						if ((state & SWT.SELECTED) != 0){
+							width += parent.showSelectedImage ? bounds.width : INTERNAL_SPACING;						
```

Comment:
@sratz We are working on extracting this logic into a separate method  `getTextPadding(...) ` for uniform padding calculation when image is not visible. This will simplify the code.

@HeikoKlare The new method `getTextPadding(...) ` will simplify it. We would be modifying the logic to increase the tab width by standard padding, for both selected and unselected tab.

Regarding the padding in line 344, it does not consider whether the image is visible or not. If image is not visible for a tab, there is very small space before the text, which makes it difficult to distinguish between tabs. Thus additional padding is to be added when image of a tab is not visible.